### PR TITLE
setup.py refreshments

### PR DESCRIPTION
**Pull request by kernc**
* Split requirements into sensible files.
* Have setuptools-git manage package data.
* Dev versioning incudes build date.
* Minimal Python 3.4.
* Travis should help keep us future-compatible (Python 3.5 & Python nightly).
* Have _CONTRIBUTING.md_,  _ISSUES_TEMPLATE.md_, and  _PULL_REQUEST_TEMPLATE.md_ GitHub will show to contributors.
* Run tests on whatever comes our of `sdist`, so we know that works.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+Coding style
+------------
+
+Roughly conform to [PEP-8] style guide for Python code. Whenever PEP-8 is
+undefined, adhere to [Google Python Style Guide].
+
+In addition, we add the following guidelines:
+* Never `import *`.
```
Comment (astaric) [inline CONTRIBUTING.md]: Except when stuff is pulled up into a parent package?

https://github.com/biolab/orange3/blob/master/Orange/classification/__init__.py

File: travis/install.sh [hunk 2]
```diff
@@ -1,11 +1,9 @@
...
 pip install -U setuptools pip wheel
-if [ ! "$(ls wheelhouse)" ]; then
-    git clone --depth=1 https://github.com/astaric/orange3-requirements wheelhouse
-else
-    echo 'Using cached wheelhouse.';
-fi
 
-pip install wheelhouse/*.whl
-pip install -r requirements.txt
+for requirements in requirements*.txt; do
```
Comment (astaric) [inline travis/install.sh]: Does order in which requirements are installed matter?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+Tests
+-----
+Ensure the tests pass:
+
+    python setup.py test
+
+If you contribute new code, make [unit tests] for it in _Orange/tests_.
+
+Prefer [doctests] for a public APIs.
```
Comment (lanzagar) [inline CONTRIBUTING.md]: for a public API
or
for public APIs

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+-------------
+Implement new features in separate topic branches:
+
+    git checkout master
+    git checkout -b my-new-feature   # spin a branch off of current branch
+
+When you are asked to make changes to your pull request, and you add the
+commits that implement those changes, [squash commits] that fit together.
+
+E.g., if you pull request looks like this:
```
Comment (lanzagar) [inline CONTRIBUTING.md]: if _your_ pull ...

**Comment by kernc** [comment]
@astaric  Please have another look at this. Seems to work much better now (might have been [Travis' fault](https://github.com/travis-ci/travis-ci/issues/5039) all along). Also, would you happen to understand why this is failing?

File: requirements-gui.txt [hunk 5]
```diff
@@ -1,2 +1,7 @@
 pyqtgraph>=0.9.10
+
+# NOTE: What is this?
```
Comment (astaric) [inline requirements-gui.txt]: orangeqt as a separate library. If I recall correctly, it is only used by parallel coordinates widget, the rest was already ported to pyqtgraph.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<!--
+Squash the commits, as appropriate. See .github/CONTRIBUTING.md for
+more information.
+
+Describe the relevant changes in the commit messages, if they need explaining.
+ 
+If the pull request introduces a new feature or an important bug fix, consider
+adding _one_ of the below tags, enclosed in square brackets, in its title:
+
+    FEATURE, FIX, DOC, WIP
```
Comment (lanzagar) [inline .github/PULL_REQUEST_TEMPLATE.md]: ENH instead of FEATURE (which is quite long)?

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+[doctests]: https://en.wikipedia.org/wiki/Doctest
+
+
+Commit messages
+---------------
+Make a separate commit for each logical change you introduce. We prefer
+short commit messages with descriptive titles. For a general format see
+[Commit Guidelines]. E.g.:
+
+> io: Fix reader for XYZ file format
```
Comment (lanzagar) [inline .github/CONTRIBUTING.md]: This example starts with a tag, but the instructions don't mention them (enough).
I think we want to emphasize this more.

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+interactively rebase the commits onto the master branch:
+
+    git rebase --interactive master
+
+and mark `fixup` or `squash` the commits that are just minor patches on
+previous commits (interactive rebase also allows you to reword and reorder
+commits). The resulting example pull request should look clean:
+
+    b432f18   Some feature I made
```
Comment (lanzagar) [inline .github/CONTRIBUTING.md]: These commit messages should represent good practice and should start with tags.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+For some example:
+
+    [FEATURE] Parallel computation of cross-validation
+    [FIX] Don't crash NaiveBayes on input containing nan values
```
Comment (lanzagar) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should we also advocate module/class tags like for commit messages?
E.g.
[FIX] naive_bayes: Don't crash on input containing nan values
or
[FIX] NaiveBayesLearner: Don't crash on input containing nan values

Which do we prefer? I like the module tags more, if the module is compact enough that the context of the change is clear. But that is not always the case... Allow the committer to decide?

File: .github/CONTRIBUTING.md (hunk 7 above)
Comment (kernc) [inline .github/CONTRIBUTING.md]: Please find [revised](https://github.com/biolab/orange3/pull/1020/files#diff-8cc7b2b0d78dd2501610391c086a8516R88).

**Comment by zidarsk8** [comment]
@lanzagar @kernc It would be nice to also update the README.md when changing the setup.

**Comment by lanzagar** [comment]
Agreed. Thanks for noticing and pointing this out.

If you already found the bits that need to be updated, you could make a pull request. Otherwise we will wait for @kernc to update the README with changes he made.

**Pull request merged**